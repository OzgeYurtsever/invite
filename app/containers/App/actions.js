import {
  ADD_INVITEE,
  ADD_INVITEE_SUCCESS,
  ADD_INVITEE_FAILURE,
  GET_COUNT,
  GET_COUNT_SUCCESS,
  GET_COUNT_FAILURE,
  GET_INVITEES,
  GET_INVITEES_SUCCESS,
  GET_INVITEES_FAILURE,
} from './constants';

const saveName = () => ({
  type: ADD_INVITEE,
});

const nameSaved = name => ({
  type: ADD_INVITEE_SUCCESS,
  payload: {
    name,
  },
});

const saveNameError = error => ({
  type: ADD_INVITEE_FAILURE,
  payload: {
    error,
  },
});

const getCount = () => ({
  type: GET_COUNT,
});

const countReceived = count => ({
  type: GET_COUNT_SUCCESS,
  payload: { count },
});

const getCountError = error => ({
  type: GET_COUNT_FAILURE,
  payload: {
    error,
  },
});

const getInvitees = () => ({
  type: GET_INVITEES,
});

const inviteesReceived = data => ({
  type: GET_INVITEES_SUCCESS,
  payload: {
    data,
  },
});

const getInviteesError = error => ({
  type: GET_INVITEES_FAILURE,
  payload: {
    error,
  },
});

export {
  saveName,
  saveNameError,
  nameSaved,
  getCount,
  countReceived,
  getCountError,
  getInvitees,
  inviteesReceived,
  getInviteesError,
};
