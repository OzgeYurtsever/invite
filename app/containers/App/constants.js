/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const ADD_INVITEE = 'invite/App/ADD_INVITEE';
export const ADD_INVITEE_SUCCESS = 'invite/App/ADD_INVITEE_SUCCESS';
export const ADD_INVITEE_FAILURE = 'invite/App/ADD_INVITEE_FAILURE';
