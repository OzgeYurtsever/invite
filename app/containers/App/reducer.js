import { fromJS } from 'immutable';

import {
  ADD_INVITEE,
  ADD_INVITEE_SUCCESS,
  ADD_INVITEE_FAILURE,
} from './constants';

const initialState = fromJS({
  loading: false,
  saving: false,
  error: false,
  count: 0,
  data: [],
});

const appReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_INVITEE:
      return state.set('saving', true).set('error', false);
    case ADD_INVITEE_SUCCESS:
      return state.set('saving', false).set('error', false);
    case ADD_INVITEE_FAILURE:
      return state.set('saving', false).set('error', action.payload.error);
    default:
      return state;
  }
};

export default appReducer;
