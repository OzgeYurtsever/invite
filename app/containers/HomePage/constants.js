export const GET_NAME = 'invite/App/GET_NAME';
