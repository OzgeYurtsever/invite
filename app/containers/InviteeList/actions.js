/*
 *
 * InviteeList actions
 *
 */

import {
  GET_INVITEES,
  GET_INVITEES_SUCCESS,
  GET_INVITEES_FAILURE,
} from './constants';

const getInvitees = () => ({
  type: GET_INVITEES,
});

const inviteesReceived = data => ({
  type: GET_INVITEES_SUCCESS,
  payload: {
    data,
  },
});

const getInviteesError = error => ({
  type: GET_INVITEES_FAILURE,
  payload: {
    error,
  },
});

export { getInvitees, inviteesReceived, getInviteesError };
