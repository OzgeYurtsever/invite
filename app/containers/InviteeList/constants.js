export const GET_INVITEES = 'invite/App/GET_INVITEES';
export const GET_INVITEES_SUCCESS = 'invite/App/GET_INVITEES_SUCCESS';
export const GET_INVITEES_FAILURE = 'invite/App/GET_INVITEES_FAILURE';
