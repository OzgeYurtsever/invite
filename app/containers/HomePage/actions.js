/*
 *
 * HomePage actions
 *
 */

import { GET_NAME } from './constants';

const getName = name => ({
  type: GET_NAME,
  payload: {
    name,
  },
});
export { getName };
