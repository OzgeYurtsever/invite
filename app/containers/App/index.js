import React from 'react';
import { Switch, Route } from 'react-router-dom';

import HomePage from 'containers/HomePage/Loadable';
import InviteeList from 'containers/InviteeList/Loadable';
import Navigation from 'components/Navigation';
import NotFoundPage from 'containers/NotFoundPage/Loadable';

import GlobalStyle from '../../global-styles';

export default function App() {
  return (
    <div>
      <Navigation />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/invitees" component={InviteeList} />
        <Route component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </div>
  );
}
