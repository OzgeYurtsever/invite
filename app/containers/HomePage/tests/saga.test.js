/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put, takeEvery } from 'redux-saga/effects';
import { ADD_INVITEE } from 'containers/App/constants';
import { nameSaved, saveNameError } from 'containers/App/actions';
import watchAddInvitee, { addName } from '../saga';

// const generator = homePageSaga();

const name = 'test name';
describe('addName Saga', () => {
  let addNameGenerator;

  beforeEach(() => {
    addNameGenerator = addName();
    const selectDescriptor = addNameGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = addNameGenerator.next(name).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the nameSaved action if the request is successful', () => {
    const response = name;
    const putDescriptor = addNameGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(nameSaved(response)));
  });

  it('should call the saveNameError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = addNameGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(saveNameError(response)));
  });
});

describe('addNameSaga Saga', () => {
  const watchAddInviteeSaga = watchAddInvitee();

  it('should start task to watch for ADD_INVITEE action', () => {
    const takeEveryDescriptor = watchAddInviteeSaga.next().value;
    expect(takeEveryDescriptor).toEqual(takeEvery(ADD_INVITEE, addName));
  });

  // it('Expect to have unit tests specified', () => {
  //   expect(true).toEqual(false);
  // });
});
