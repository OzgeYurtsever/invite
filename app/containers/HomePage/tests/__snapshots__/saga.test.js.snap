// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addName Saga should call the saveNameError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`addName Saga should call the saveNameError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/api/invitees",
      Object {
        "body": "{\\"name\\":\\"test name\\"}",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`addName Saga should dispatch the nameSaved action if the request is successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`addName Saga should dispatch the nameSaved action if the request is successful 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/api/invitees",
      Object {
        "body": "{\\"name\\":\\"test name\\"}",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
